### ISSUE TITLE
Config file is not recognized; autocorrect not working

### ISSUE THREAD

**Issue by Den-Rimus** [detekt/detekt#5169]
## Expected Behavior
1) Detekt should pick up specified `*.yml` file with config
2) Autocorrectable issues should be corrected on detekt run

## Observed Behavior
1) config ignored
2) nothing is corrected

## Steps to Reproduce
Reproducible repo setup [here](https://github.com/Den-Rimus/DetektConfigAutocorrectReproduce)
1) Observe repo git commits (after initial)
2) run `./gradlew detekt` from command line in project's root dir
3a) (problem 1) Changes to detekt config made in commits are not applied.
3b) (problem 2) Nothing autocorrected (look at empty method, trailing spaces, no new line at the end of file)

To make sure: AndroidStudio's settings about trailing spaces and ensuring new line at the end of file are off - same as relevant  `.editorconfig` parameters.

## Context
Trying to do the right thing and setup detekt in our fancy new app project to keep code nice and clean from very beginning.

## Your Environment

* Version of detekt used: 1.21.0
* Version of Gradle used (if applicable): AGP 7.2.1, Gradle 7.3.3
* Gradle scan link (add `--scan` option when running the gradle task): https://scans.gradle.com/s/dgzowyx3ixt2g
* Operating System and version: MacOS 12.4
* Link to your project (if it's a public repository): https://github.com/Den-Rimus/DetektConfigAutocorrectReproduce

P. S.: I know that bug of this severity is too huge to be true, but I've double-, triple- and quadruple-checked and re-checked all setup in my initial repo (where I am trying to use detekt) and then reproduced it in given public dummy-repo. It is either broken or something is terribly wrong with your documentation or, third option, I am so stupid that shouldn't even be able to breathe (which seems not to be the case since I've made it this far to my age 😄 ).

**Comment by 3flex** [comment]
You're applying the detekt config to the root project only. In `build.gradle.kts` you have:
```kotlin
detekt {
   config = files("config/detekt/custom_detekt.yml")
   autoCorrect = true
}
```

Which is only being applied to the root project, but none of the subprojects. You can instead apply the config in the `allprojects {}` block like so:
```kotlin
allprojects {
    apply(plugin = "io.gitlab.arturbosch.detekt")
    detekt {
        config = rootProject.files("config/detekt/custom_detekt.yml")
        autoCorrect = true
    }

// ...
}
```

Only rules from the `detekt-formatting` rule set support autocorrect.

Let us know if there are areas of the documentation that should be updated to make this clearer, or if anything in the docs is incorrect.

**Comment by Den-Rimus** [comment]
Ok, that suggestion worked. I've never seen actual mentioning of which build.gradle(.kts) file (root project or sumbodule(s)) should have detekt plugin apply line or detekt configuring closure dependency for plugin with formatting. Maybe that could be improved.

> Only rules from the detekt-formatting rule set support autocorrect.

How do I find this rule set? How do I configure it? So far (with current setup as in given repo) autoCorrect flag does nothing (although I've applied all setup necessary as per detekt documentation). Could you give me one small push again with this?

**Comment by Den-Rimus** [comment]
Nevermind. It is generated with corresponding task but only when formatting plugin dependency is applied. This should definitely be in docs.

Thank you for support and great job on this tool! 👍
